### PR TITLE
APIM: Add v2 SKU

### PR DESCRIPTION
**Pull request by ross-p-smith**
This PR adds v2 SKU of API Service

Relates to #2747

**Special notes for your reviewer**:

**How does this PR make you feel**:
gif

**If applicable**:
- [x] this PR contains documentation
- [x] this PR contains tests

### CODE REVIEW

File: v2/internal/controllers/crd_apimanagement_20230501preview_test.go [hunk 1]
```diff
@@ -0,0 +1,497 @@
...
+
+func Test_ApiManagement_20230501preview_CRUD(t *testing.T) {
+	t.Parallel()
+
+	tc := globalTestContext.ForTest(t)
+	rg := tc.NewTestResourceGroup()
+
+	tc.CreateResourceAndWait(rg)
+
+	// The v2 SKU has a much quicker start up time.
```
Comment (matthchr) [inline v2/internal/controllers/crd_apimanagement_20230501preview_test.go:29]: TBH I'd be OK with deleting the other (slower) test if you think that makes sense.

Comment (matthchr) [reply v2/internal/controllers/crd_apimanagement_20230501preview_test.go:29]: Could probably do in a separate PR (or I can do it once this PR merges if you're busying doing other things)

**Review by matthchr — APPROVED** [review]
This looks great to me.

**Review by super-harsh — APPROVED** [review]

File: v2/internal/controllers/crd_apimanagement_20230501preview_test.go (hunk 1 above)
Comment (ross-p-smith) [reply v2/internal/controllers/crd_apimanagement_20230501preview_test.go:29]: Let's tidy up in a separate PR and then it's explicit what and why we are doing it

Comment (ross-p-smith) [reply v2/internal/controllers/crd_apimanagement_20230501preview_test.go:29]: Added #3695

**Comment by matthchr** [comment]
Our CI SP expired - once that's fixed we'll merge this.

**Comment by tomkerkhove** [comment]
Oh the certainties in life!

**Pull request merged**